### PR TITLE
Th2 3616 - array support

### PR DESCRIPTION
**Pull request by Topru333**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+    dependencies {
+        classpath "com.exactpro.th2:box-descriptor-generator:0.1.10"
+    }
+}
+
+
 plugins {
     id 'java-library'
     id 'application'
+    id 'signing'
```
Comment (OptimumOpium) [inline build.gradle:17]: We don't publish any artifacts. Do we need this plugin?

File: src/main/java/com/exactpro/th2/codec/csv/cfg/CsvCodecConfiguration.java [hunk 2]
```diff
@@ -39,6 +39,10 @@ public class CsvCodecConfiguration {
     @JsonPropertyDescription("Display name for the root event sent to the event store")
     private String displayName = "CodecCsv";
 
+    @JsonProperty("validate-length")
+    @JsonPropertyDescription("Set to validate length of columns or not")
+    private Boolean validateLength = true;
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/codec/csv/cfg/CsvCodecConfiguration.java:44]:
```suggestion
    private boolean validateLength = true;
```

File: src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java [hunk 3]
```diff
@@ -175,12 +175,32 @@ private void decodeCsvData(Collection<ErrorHolder> errors, MessageGroup.Builder
...
             int rowLength = strings.length;
             for (int i = 0; i < headerLength && i < rowLength; i++) {
-                builder.putFields(header[i], ValueUtils.toValue(strings[i]));
+                int extraLength = getGetExtraLength(i, header, rowLength);
+                if (extraLength == 0) {
+                    builder.putFields(header[i], ValueUtils.toValue(strings[i]));
+                } else {
+                    String[] values = new String[extraLength + 1];
+                    System.arraycopy(strings, i, values, 0, extraLength + 1);
+                    builder.putFields(header[i], ValueUtils.toValue(values));
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java:184]: I am not sure if we have the overload for arrays in `toValue` function. Can you check it and if we don't have overload - use List instead?

File: README.md [hunk 4]
```diff
@@ -63,6 +66,11 @@ spec:
 
 ### 3.1.0
 
++ new length validation parameter
++ array support, to specify array need to have empty header for each value
```
Comment (OptimumOpium) [inline README.md:70]: Please, add example of the CSV and the result message

File: src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java (hunk 3 above)
Comment (Topru333) [reply src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java:184]: > fun Array<*>.toValue(): Value = iterator().toValue()

we have it

File: README.md [hunk 5]
```diff
@@ -11,6 +11,26 @@ Otherwise, the default header will be used for decoding the rest of data.
 
 If no data was decoded from raw message, the message will be skipped, and an error event will be reported.
 
+## Example
+
```
Comment (OptimumOpium) [inline README.md:15]: Please, also add examples for normal csv format without empty headers

**Review by OptimumOpium — APPROVED** [review]

**Pull request merged**